pkg/synthetictests/allowedalerts/job_run_data.go

```diff
@@ -25,7 +25,7 @@ func WriteAlertDataForJobRun(artifactDir string, _ monitorapi.ResourcesMap, even
 
```

Comment:
Not really related to your current change but more of a long term plan question (which I believe you are working on).  Can we work to at least co-locate [AlertKey](https://github.com/neisw/origin/blob/ebf4f252b21666844904e021e7feb6290e9655c1/pkg/synthetictests/allowedalerts/job_run_data.go#L54) , [AlertState](https://github.com/openshift/origin/blob/ebf4f252b21666844904e021e7feb6290e9655c1/pkg/synthetictests/allowedalerts/basic_alert.go#L37) along with [AlertLevel](https://github.com/openshift/origin/blob/ebf4f252b21666844904e021e7feb6290e9655c1/pkg/synthetictests/allowedalerts/job_run_data.go#L81) and comment on how each is 'different' such that we require separate types.  Preferably in [alert_types](https://github.com/openshift/origin/blob/ebf4f252b21666844904e021e7feb6290e9655c1/pkg/synthetictests/historicaldata/alert_types.go#L21)

Reply:
Might be too much of a reach to try to tie in a common type to [prometheus/helpers](https://github.com/openshift/origin/blob/1e39f421a95fc8510dfc25780fe3cbfccf7c0b26/test/extended/util/prometheus/helpers.go#L135) and [AlertDataKey](https://github.com/openshift/origin/blob/ebf4f252b21666844904e021e7feb6290e9655c1/pkg/synthetictests/historicaldata/alert_types.go#L21)

Just seems like so much commonality dispersed throughout the code.

Reply:
Mmm MetricCondition is for whitelisting alerts, which is also used for any alert that got it's own test. AlertLevel is largely going away on both this MetricCondition and on AlertDataKey, alerts only have one Level, we don't need to key on it. That will help trim down a little. 

I'd like to instead just iterate all the per alert tests and say "hey do you match this interval", but we'd still need MetricCondition for explicitly whitelisting alerts that are allowed to happen. I think I can improve this in round three though, let me give it some thought and work on it there.

Reply:
That's all I was going for was what can be removed / consolidated in the future.

Name & Namespace seem to be paired together throughout.  Level & State look very similar.  Good to hear AlertLevel is going away, that should tighten things up some.